### DISCUSSION TITLE
checkDependecies() is missing package dependencies (@nx/js:tsc)

### DISCUSSION THREAD

**Discussion by sebastian-villate** [nrwl/nx#17692] [General]
Hey everyone, 
I'm currently working with [@nx/js:tsc](https://nx.dev/packages/js/executors/tsc), but I realized that when it generates the package.json it is missing a dependency. 

The package in question is packageA, and the missing dependency is on packageB. The folder structure is 

```
Workspace Root/
├─ packages/
│  ├─ packageA/
│  ├─ packageB/

```

and when I set up the target I am only filling the outputPath, main and tsconfig

I went through the source code and tracked it down to the checkDependencies function which is returning an incomplete list of dependencies. Has anyone had a similar issue?

**Comment by lazee** [comment]
package resolving is some tricky stuff, and yes, there has been issues with this. My recommendation is to setup a repo in github that demonstrates the problem. Then maybe we can track down the problem together.
